network/network.go

```diff
@@ -196,10 +204,21 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 			start := time.Now()
 
 			c.logger.Log("level", "info", "message", "dialing host", "host", host, "scrapeID", c.scrapeID)
-			conn, err := c.dialer.Dial("tcp", host)
-			if err != nil {
-				c.logger.Log("level", "error", "message", "could not dial host", "host", host, "scrapeID", c.scrapeID, "stack", microerror.Stack(err))
-				c.dialErrorCount.WithLabelValues(host).Inc()
+			conn, dialErr := c.dialer.Dial("tcp", host)
+			if dialErr != nil {
+				podExists, err := c.podExists(host, pods)
+				if err != nil {
+					c.logger.Log("level", "error", "message", "unable to check if host exists", "host", host, "scrapeID", c.scrapeID, "stack", microerror.Stack(err))
```

Comment:
In this case, where the dial failed and the Pod lookup also failed, should this still be counted as a dial error?

Reply:
I would say so.

Reply:
I agree. Added